packages/plugin-patch/sources/commands/patchCommit.ts

```diff
@@ -43,15 +43,15 @@ export default class PatchCommitCommand extends BaseCommand {
 
     const originalPath = await patchUtils.extractPackageToDisk(locator, {cache, project});
 
-    const originalPathN = npath.fromPortablePath(originalPath);
-    const patchedPathN = npath.fromPortablePath(folderPath);
+    const originalPathN = npath.fromPortablePath(originalPath).replace(/\\/g, `/`);
+    const patchedPathN = npath.fromPortablePath(folderPath).replace(/\\/g, `/`);
 
     let {stdout} = await execUtils.execvp(`git`, [`diff`, `--no-index`, originalPathN, patchedPathN], {
       cwd: this.context.cwd,
     });
 
-    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(originalPathN), `g`), ``);
-    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(patchedPathN), `g`), ``);
+    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(`a/${originalPathN}/`), `g`), `a/`);
+    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(`b/${patchedPathN}/`), `g`), `b/`);
```

Comment:
Remains the problem of `a//home/foo/file.txt` (also a test would be useful, given how often we get it wrong 😅).

Reply:
Yup, just merged in master so I could switch branches without rewriting the entire cache